addons/account/models/ir_attachment.py

```diff
@@ -11,9 +12,78 @@
 _logger = logging.getLogger(__name__)
 
 
+def split_etree_on_tag(tree, tag):
+    """ Split an etree that has multiple instances of a given tag into multiple trees
+    that each have a single instance of the tag. """
+    tree = deepcopy(tree)
+    nodes_to_split = tree.findall(f'.//{tag}')
+
+    # Remove all nodes with the tag
+    parent_node = nodes_to_split[0].getparent()
+    for node in nodes_to_split:
+        parent_node.remove(node)
+
+    # Create a new tree for each node
+    trees = []
+    for node in nodes_to_split:
+        parent_node.append(node)
+        trees.append(deepcopy(tree))
+        parent_node.remove(node)
+    return trees
+
+
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    import_type = fields.Char(
+        string="File type for import",
+        help="Technical field, used by the import framework to call the appropriate decoder.",
+        compute='_compute_import_type_and_priority',
+    )
+    import_priority = fields.Integer(
+        string="Import priority",
+        help="Technical field indicating the priority with which this attachment should be decoded on an invoice.",
+        compute='_compute_import_type_and_priority',
+    )
+    xml_tree = fields.Binary(
+        string="XML tree for import",
+        help="Technical field containing the lxml.etree of the attachment, to avoid needing to recompute it.",
+        compute='_compute_xml_tree',
+    )
+    root_attachment_id = fields.Many2one(
+        string="Root attachment",
+        help="Technical field indicating the attachment from which this embedded attachment was extracted.",
+        comodel_name='ir.attachment',
+        store=False,  # only needed during invoice import, so we keep it cache-only.
+    )
+    origin_attachment_id = fields.Many2one(
+        string="Origin attachment",
+        help="Technical field which is computed as `origin_attachment_id = root_attachment_id or self`",
+        comodel_name='ir.attachment',
+        compute='_compute_origin_attachment_id',
+    )
```

Comment:
put all of that in dictionaries. You don't want to bloat such low-level model with very specific stuff only for the import